test/integration/conftest.py

```diff
@@ -141,11 +142,23 @@ def db(mongo_client):
 @pytest.fixture(scope="class")
 def empty_db(db):
     PRESERVED_COLLECTIONS = ["groups", "organizations", "allowedDomains"]
+    admin_found = False
+    admin_email = "admin@example.com"
+    if "TEST_ADMIN_EMAIL" in os.environ:
+        admin_email = os.environ["TEST_ADMIN_EMAIL"]
     for name in db.list_collection_names():
         if name.startswith("system."):
             continue
         if name in PRESERVED_COLLECTIONS:
             continue
+        if name == "users":
```

Comment:
I don't understand why you "special case" the `users` collection here. Is there any reason you can't explicitly check whether the admin user is already present (e.g. `db.users.find({email: admin_email})`) and insert it if it isn't present before this loop, and then just add `users` to `PRESERVED_COLLECTIONS`?

Reply:
I left the user account creation in the `conservator()` fixture (defined just after `empty_db()`), since it needs to collect the API key to return a `Conservator` object there anyhow.

Reply:
I didn't notice it was in a separate function. In any case:
 - There is no reason to clear out the `users` collection - I don't think there are any integration tests that rely on it being empty, and it's an annoyance on developer machines if you're trying to log in using a pre-created admin user
 - The way the admin user is searched for and created in the `conservator()` fixture seems sub-optimal. The logic, IMHO, should be:
     - Search for user by email address
         - If it's there, check that it has the expected API key
             - If it does, break
             - If it doesn't, update the record to have the expected API key
         - If it's not there, create it

Reply:
Also, I discovered that there was already an account with the default `admin@example.com` once the Kind cluster is up and running, but it has no `apiKey` field in it, so it can be a bit more complicated than the `db.users.find({email: admin_email})` query.

Reply:
Implemented the proposed logic change for the conservator() fixture.